simulator/risc_v/riscv_decoder.h

```diff
@@ -16,8 +16,17 @@
 enum class Reg : uint8
 {
     RS1, RS2, RD,
+    RS1_, RS2_, RD_,
     CSR, SEPC, MEPC,
-    ZERO, RA
+    ZERO, RA, SP
+};
+
+enum ImmediateType
```

Comment:
Could you please use char for non-compressed immediates and add a new enum for compressed immediates?